lib/android-helpers.js

```diff
@@ -419,7 +419,13 @@ helpers.pushSettingsApp = async function (adb, throwError = false) {
 
 helpers.pushUnlock = async function (adb) {
   logger.debug("Pushing unlock helper app to device...");
-  await adb.installOrUpgrade(unlockApkPath, UNLOCK_HELPER_PKG_ID, true);
+
+  try {
```

Comment:
we could extract settings and unlock helper install into a separate method to avoid unnecessary code duplication

Also, in the warning message we can state the possibility to uninstall the tool manually from the device in order to fix the problem.